qiskit_experiments/framework/base_experiment.py

```diff
@@ -30,6 +31,33 @@
 from qiskit_experiments.framework.configs import ExperimentConfig
 
 
+def cached_method(method):
+    """Decorator to cache the return value of a BaseExperiment method.
+
+    This stores the output of a method in the experiment object instance
+    in a `_cache` dict attribute. Note that the value is cached only on
+    the object instance method name, not any values of its arguments.
+
+    The cache can be cleared by calling :meth:`.BaseExperiment.cache_clear`.
+    """
+
+    @wraps(method)
+    def wrapped_method(self, *args, **kwargs):
+        name = f"{type(self).__name__}.{method.__name__}"
```

Comment:
Class name is necessary?

Reply:
This was so if subclasses override a cached method, the supers method can still be cached

Reply:
Make sense.